core/arch/arm/arm.asm

```diff
@@ -181,24 +167,30 @@ GLOBAL_LABEL(dr_call_on_clean_stack:)
 #ifdef DR_APP_EXPORTS
         DECLARE_EXPORTED_FUNC(dr_app_start)
 GLOBAL_LABEL(dr_app_start:)
-        push     {lr}
-        vstmdb   sp!, {d16-d31}
-        vstmdb   sp!, {d0-d15}
-        mrs      REG_R0, cpsr /* r0 is scratch */
-        push     {REG_R0}
-        /* We can't push all regs w/ writeback */
-        stmdb    sp, {REG_R0-r15}
-        str      lr, [sp, #(PRIV_MCXT_PC_FROM_SIMD+4)] /* +4 b/c we pushed cpsr */
-        /* we need the sp at function entry */
-        mov      REG_R0, sp
-        add      REG_R0, REG_R0, #(PRIV_MCXT_SIMD_SIZE + 8) /* offset simd,cpsr,lr */
-        str      REG_R0, [sp, #(PRIV_MCXT_SP_FROM_SIMD+4)] /* +4 b/c we pushed cpsr */
-        sub      sp, sp, #(PRIV_MCXT_SIZE-PRIV_MCXT_SIMD_SIZE-4) /* simd,cpsr */
-        mov      REG_R0, sp
+#if (PRIV_MCXT_SIZE + 8) % 8 != 0
+#    error Stack should be 8-byte aligned.
```

Comment:
Should this say "priv_mcontext_t size should be 8-byte aligned"

Reply:
... that would be clearer ...